resolve/config.go

```diff
@@ -64,7 +64,12 @@ func (o regexpOverrideSpec) matches(imp ImportSpec, lang string) bool {
 }
 
 func (o regexpOverrideSpec) resolveRegexpDep(imp ImportSpec) label.Label {
-
+	// If "$" is found in the dependency string, then backreferences exist and
+	// ReplaceAllString() should be run on the string to substitute in the
+	// correct replacement strings to build the label.
+	if !strings.Contains(o.dep.String(), "$") {
+	    return o.dep
+	}
```

Comment:
This follows the [same logic](https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/regexp/regexp.go;l=574) found in the `regex` Go stdlib to determine if backreferences exist.

I'm not entirely sure these are really backreferences, as the Go stdlib doesn't support backreferences, but it at least is indicating that the user expects regex to perform string replacement.  I think it is more accurately referred to as `Regexp.Expand` in this library.

Reply:
In theory `$` could be escaped, but since `$` is pretty unlikely to make it into a label, let alone into a replace pattern, I think that's okay for now.